_rules/iframe-not-focusable-has-no-interactive-content-akn7bn.md

```diff
@@ -0,0 +1,104 @@
+---
+id: akn7bn
+name: iframe with negative tabindex has no interactive elements
+rule_type: atomic
+description: |
+  This rule checks that `iframe` elements with a negative `tabindex` attribute value contain no interactive elements.
+accessibility_requirements:
+  wcag20:2.1.1: # Keyboard (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+  wcag-technique:G202: # Ensuring keyboard control for all functionality
+    forConformance: false
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any `iframe` element that has a negative number as a `tabindex` [attribute value][].
+
+## Expectation
+
+None of the test target's [nested browsing context][] includes elements that are [visible][] and part of the [sequential focus navigation][]. An element is "included" in a [nested browsing context][] if its [owner document][] is the [container document][] of the [nested browsing context][].
```

Comment:
Linked to previous comment, I understand the expectation as not going recursively into nested `iframe`, because "owner document" only look at the parent, not the top-level ancestor.
I don't know if this is intentional or not.

Reply:
It's not recursive, no. Because `iframe` elements are themselves focusable, an empty `iframe` inside your `iframe[tabindex="-1"]` would fail this rule unless it too had a negative tabindex. That is captured in the assumption. I think that's reasonable. Iframes need to be focusable in order to allow scrolling with keyboard.

Reply:
🤔 
```html
<iframe tabindex="-1" srcdoc="<iframe tabindex='-1' srcdoc='<a href=…>W3C</a>' />" />
```
(not filling the `href` to avoid quote protection nightmare)
Then the link is not keyboard accessible (the SC is not satisfied) but the rule would pass because it only looks at the outer `iframe` and is happy to see that none of its elements are part of sequential focus navigation.

And I don't think this is caught by the assumption since the iframe (both inner and outer) do indeed provide functionality.

This is not a blocker, though since it means the rule passes too often, which is usual.

Reply:
Ah no, my bad. The inner iframe should be applicable and fail.
It is not completely explicit in the applicability, however…